### DISCUSSION TITLE
Default sorted column in AnalyticalTable

### DISCUSSION THREAD

**Discussion by SvetoslavZ99** [UI5/webcomponents-react#6383] [Q&A] [answered]
Hello,

I am using AnalyticalTable component (v1) and have provided the "filterable" prop.

Everything is working as expected, when I click on one of the table columns, I can properly filter in asc or dsc order.

The thing is, I want my first column to be in asc order sorted when I load my page, and the icon to be visible, so the user can know that this is the default for this table. (in the images example taken from the docs, is the Name column) 

Is there a way to set it up properly, maybe I am missing something while I create my table columns? I couldnt find an example in the documentation with pre-sorted column. 

<img width="893" alt="Screenshot 2024-09-19 at 19 49 21" src="https://github.com/user-attachments/assets/9a8d8835-7fe6-4693-992c-3e6295184563">
<img width="889" alt="Screenshot 2024-09-19 at 19 49 29" src="https://github.com/user-attachments/assets/affc86da-3412-41af-b184-2fa289c94679">

**Comment by Lukas742** [comment, marked as answer]
Hi @SvetoslavZ99 

you can use the `reactTableOptions.initialState` object to achieve this. 

```jsx
      <AnalyticalTable
        {...otherProps}
        reactTableOptions={{
          initialState: {
            columnOrder: ['age', 'name'],
            filters: [{ id: 'age', value: '27' }],
            groupBy: ['age'],
            hiddenColumns: ['name'],
            selectedRowIds: { 2: true, 5: true, '18.3': true },
            sortBy: [
              {
                id: 'age',
                desc: true,
              },
            ], // <-- this is the option you can use
            expanded: {
              '18': true,
              '18.3': true,
            },
          },
        }}
      />
```

https://stackblitz.com/edit/ui5wcr-at-initial-state?file=src%2FApp.tsx,src%2FdataTree.json

**Reply by SvetoslavZ99** [reply]
Thanks a lot! That is pretty clean solution!
